### PR TITLE
Passkey miscellaneous checks part 1

### LINKED ISSUE #2031: [Feature]: Passkey- implement misc validations
### Feature Description

Verify and implement all of other miscellaneous verification checks for the transaction

## Acceptence Criterea
- verify and implement to (re-use) all of the remaining [SignedExtra](https://github.com/frequency-chain/frequency/blob/d25b833d77a9968fa5d01d7e21e948e4d0ee16d9/runtime/frequency/src/lib.rs#L260) extensions.
- Add tests
- Add benchmarks

## Notes
It is preferable if we can reuse these implementations instead of fresh implementations

### Searched for Related Issues

- [X] I have done a search for related issues and either found none, or noted them

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goal
The goal of this PR is to incorporate more checks for passkey extrinsic inline with other SignedExtras
- [x] Non Zero Sender Check
- [x] Spec Check
- [x] Tx version check
- [x] Era check
- [x] Genesis check

Closes #2031 

# Discussion

# Checklist
- [ ] Chain spec updated
- [ ] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [ ] Design doc(s) updated
- [ ] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

File: pallets/passkey/src/lib.rs [hunk 1]
```diff
@@ -337,38 +353,72 @@ where
...
+		let runtime_call: <T as frame_system::Config>::RuntimeCall =
+			<T as frame_system::Config>::RuntimeCall::from(self.1.clone());
+		let who = self.0.clone();
 
-		Ok(ValidTransaction { priority, ..Default::default() })
+		let non_zero_sender_check = frame_system::CheckNonZeroSender::<T>::new();
+		let spec_version_check = frame_system::CheckSpecVersion::<T>::new();
+		let tx_version_check = frame_system::CheckTxVersion::<T>::new();
+		let genesis_hash_check = frame_system::CheckGenesis::<T>::new();
+		let era_check = frame_system::CheckEra::<T>::from(Era::immortal());
```
Comment (saraswatpuneet) [inline pallets/passkey/src/lib.rs:388]: Order of check is kept same as in [SignedExtras](https://github.com/frequency-chain/frequency/blob/d25b833d77a9968fa5d01d7e21e948e4d0ee16d9/runtime/frequency/src/lib.rs#L260)

File: pallets/passkey/src/lib.rs [hunk 2]
```diff
@@ -350,3 +362,63 @@ where
...
+
+impl<T: Config + Send + Sync> FrameSystemChecks<T>
+where
+	<T as frame_system::Config>::RuntimeCall:
+		From<Call<T>> + Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
+{
+	/// Validates the transaction fee paid with tokens.
+	pub fn pre_dispatch(&self) -> Result<(), TransactionValidityError> {
+		let info = &self.1.get_dispatch_info();
+		let len = self.0.using_encoded(|c| c.len());
```
Comment (aramikm) [inline pallets/passkey/src/lib.rs:379]: Shouldn't this be the encoded length of the call?

File: pallets/passkey/src/lib.rs [hunk 3]
```diff
@@ -350,3 +362,63 @@ where
...
+		spec_version_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		tx_version_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		genesis_hash_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		era_check.pre_dispatch(&who, &runtime_call, info, len)
+	}
+
+	/// Validates the transaction fee paid with tokens.
+	pub fn validate(&self) -> TransactionValidity {
+		let info = &self.1.get_dispatch_info();
+		let len = self.0.using_encoded(|c| c.len());
```
Comment (aramikm) [inline pallets/passkey/src/lib.rs:400]: same in here?

File: pallets/passkey/src/lib.rs (hunk 2 above)
Comment (saraswatpuneet) [reply pallets/passkey/src/lib.rs]: you are correct, good catch

**Review by aramikm — APPROVED** [review]

File: pallets/passkey/src/lib.rs [hunk 4]
```diff
@@ -350,3 +362,63 @@ where
...
+		let spec_version_check = frame_system::CheckSpecVersion::<T>::new();
+		let tx_version_check = frame_system::CheckTxVersion::<T>::new();
+		let genesis_hash_check = frame_system::CheckGenesis::<T>::new();
+		let era_check = frame_system::CheckEra::<T>::from(Era::immortal());
+
+		non_zero_sender_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		spec_version_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		tx_version_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		genesis_hash_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		era_check.pre_dispatch(&who, &runtime_call, info, len)
```
Comment (enddynayn) [inline pallets/passkey/src/lib.rs:394]: I just wanted to double check if we are missing a `?` operator here?

**Review by enddynayn — APPROVED** [review]
👍 great!

**Comment by enddynayn** [comment]
This is great. I wonder if we need to add some test here?

File: pallets/passkey/src/lib.rs (hunk 4 above)
Comment (saraswatpuneet) [reply pallets/passkey/src/lib.rs:404]: No since this returns from this point 
> I just wanted to double check if we are missing a ? operator here?

**Review by mattheworris — APPROVED** [review]
Looks good! 
🚢 it!

**Pull request merged**